plugins/inputs/beat/beat_test.go

```diff
@@ -0,0 +1,225 @@
+package beat
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+var beatTest = NewBeat()
+
+var (
+	beat6StatsAccumulator testutil.Accumulator
+)
+
+func Test_BeatStats(test *testing.T) {
+	//beat6StatsAccumulator.SetDebug(true)
+	fakeServer := httptest.NewUnstartedServer(
+		http.HandlerFunc(
+			func(writer http.ResponseWriter, request *http.Request) {
+				writer.Header().Set("Content-Type", "application/json")
+				if request.URL.String() == suffixInfo {
+					fmt.Fprintf(writer, "%s", string(beat6Info))
+				} else if request.URL.String() == suffixStats {
+					fmt.Fprintf(writer, "%s", string(beat6Stats))
+				} else {
+					test.Logf("Unkown URL: " + request.URL.String())
+				}
+			},
```

Comment:
The test-data should move to a JSON file you load and dump over HTTP. In the current schema you rely on the compiler to detect that the consts can be compiled out which is... not so nice. :-)

Look at other current test schemes (e.g. into the starlark processor or the `inputs/rabbitmq` tests) to get an idea how this could look like.

Reply:
will fix